### PR TITLE
[DOC] Regression api

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### What does this implement/fix? Explain your changes.

This adds CNNRegressor, TapNetRegressor and RocketRegressor to the regression api. I have also expanded the docstring for TapNetRegressor to be a bit more descriptive and added and init to the regression.deep_learning package.  CNNRegressor and TapNetRegressor should be private, but that would require deprecation, and is beyond the scope of this [DOC] PR.

### CODE REVIEW

**Review by MatthewMiddlehurst — DISMISSED** [review]
API changes are done in the style of #3742. Should probably wait for that first, but I approve of the changes.

File: docs/source/api_reference/regression.rst [hunk 1]
```diff
@@ -5,9 +5,14 @@ Time series regression
...
-    :no-members:
-    :no-inherited-members:
+All current sktime Regressors can be listed as follows
+
+```
+from sktime.registry import all_estimators
+cls = all_estimators(estimator_types="regressor")
+names = [i for i, _ in cls]
+print(names)
+```
```
Comment (MatthewMiddlehurst) [inline docs/source/api_reference/regression.rst:15]: This does not seem to be rendering as intended, it's all on one line. i.e. `from sktime.registry import all_estimators cls = all_estimators(estimator_types="regressor") names = [i for i, _ in cls] print(names)`.

File: docs/source/api_reference/regression.rst [hunk 2]
```diff
@@ -30,3 +47,31 @@ Interval-based
...
+
+    BaseRegressor
+
+.. currentmodule:: sktime.regression.deep_learning
+
+.. autosummary::
+    :toctree: auto_generated/
+    :template: class.rst
+
+    BaseDeepRegressor
```
Comment (MatthewMiddlehurst) [inline docs/source/api_reference/regression.rst:77]: Both of these seem to be missing also. Maybe the whitespace is necessary?

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
Some small bits.

File: docs/source/api_reference/regression.rst (hunk 1 above)
Comment (TonyBagnall) [reply docs/source/api_reference/regression.rst]: how is it now? cant really tell in the preview

File: docs/source/api_reference/regression.rst (hunk 2 above)
Comment (TonyBagnall) [reply docs/source/api_reference/regression.rst]: should be fixed

**Review by MatthewMiddlehurst — APPROVED** [review]
I made a push fixing the module import, both base classes now show on the page.

**Pull request merged**